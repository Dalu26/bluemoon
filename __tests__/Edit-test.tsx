import 'react-native';
import React from 'react';
import Edit from '../src/screens/more/edit';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const testProps = (props: Object) => ({
  navigation: {
    state: {params: {}},
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn(),
  },
  route: {
    params: {item: null},
  },
  ...props,
});
let props = testProps({});

it('renders correctly', () => {
  const edit= renderer.create(<Edit {...props} />)
  expect(edit).toMatchSnapshot();
});
