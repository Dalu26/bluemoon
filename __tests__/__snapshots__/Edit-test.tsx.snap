// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "rgba(98, 47, 181, 1)",
      "flex": 1,
    }
  }
  testID="edit"
>
  <BVLinearGradient
    colors={
      Array [
        4284624821,
        4279963446,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "flex": 1,
        "marginTop": 23.5,
        "paddingHorizontal": 27,
        "paddingTop": 53.5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 79,
            "width": "100%",
          },
          Object {
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#e1d5f5",
              "borderRadius": 9.5,
              "height": 55.5,
              "width": 67.5,
            },
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï„¯
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Poppins-Regular",
              "fontSize": 26,
            },
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 35,
                "marginLeft": 19.5,
              },
              Object {
                "fontFamily": "Poppins-Bold",
              },
            ],
          ]
        }
      />
    </View>
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      showsVerticalScrollIndicator={false}
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Poppins-Regular",
                "fontSize": 26,
              },
              Array [
                Object {
                  "fontFamily": "Poppins-Bold",
                },
                Object {
                  "color": "#ffffff",
                  "fontSize": 44,
                  "marginVertical": 31.5,
                },
              ],
            ]
          }
        >
          Edit & Save an item.
        </Text>
        <View
          style={
            Array [
              Object {},
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              autoFocus={true}
              maxLength={50}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Name"
              placeholderTextColor="#ffffff"
              style={
                Array [
                  Object {
                    "backgroundColor": "#2B1A6C",
                    "borderRadius": 28.5,
                    "borderWidth": 2,
                    "color": "#ffffff",
                    "fontSize": 28,
                    "minHeight": 102.5,
                    "paddingHorizontal": 35,
                    "width": "100%",
                  },
                  Object {
                    "marginVertical": 8,
                  },
                  Object {
                    "fontFamily": "Poppins-Regular",
                  },
                  Object {
                    "borderColor": "transparent",
                  },
                ]
              }
              textAlignVertical="center"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="numeric"
              maxLength={8}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="$NaN"
              placeholderTextColor="#ffffff"
              style={
                Array [
                  Object {
                    "backgroundColor": "#2B1A6C",
                    "borderRadius": 28.5,
                    "borderWidth": 2,
                    "color": "#ffffff",
                    "fontSize": 28,
                    "minHeight": 102.5,
                    "paddingHorizontal": 35,
                    "width": "100%",
                  },
                  Object {
                    "marginVertical": 8,
                  },
                  Object {
                    "fontFamily": "Poppins-Regular",
                  },
                  Object {
                    "borderColor": "transparent",
                  },
                ]
              }
              textAlignVertical="center"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              keyboardType="numeric"
              maxLength={8}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Total Stock"
              placeholderTextColor="#ffffff"
              style={
                Array [
                  Object {
                    "backgroundColor": "#2B1A6C",
                    "borderRadius": 28.5,
                    "borderWidth": 2,
                    "color": "#ffffff",
                    "fontSize": 28,
                    "minHeight": 102.5,
                    "paddingHorizontal": 35,
                    "width": "100%",
                  },
                  Object {
                    "marginVertical": 8,
                  },
                  Object {
                    "fontFamily": "Poppins-Regular",
                  },
                  Object {
                    "borderColor": "transparent",
                  },
                ]
              }
              textAlignVertical="center"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              multiline={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Description"
              placeholderTextColor="#ffffff"
              style={
                Array [
                  Object {
                    "backgroundColor": "#2B1A6C",
                    "borderRadius": 28.5,
                    "borderWidth": 2,
                    "color": "#ffffff",
                    "fontSize": 28,
                    "minHeight": 102.5,
                    "paddingHorizontal": 35,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "marginVertical": 8,
                    },
                    Object {
                      "height": 189.5,
                    },
                  ],
                  Object {
                    "fontFamily": "Poppins-Regular",
                  },
                  Object {
                    "borderColor": "transparent",
                  },
                ]
              }
              textAlignVertical="top"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#7F4CD2",
                "borderRadius": 39.5,
                "display": "flex",
                "height": 155,
                "justifyContent": "center",
                "marginTop": 63,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Poppins-Regular",
                      "fontSize": 26,
                    },
                    Array [
                      Object {
                        "fontFamily": "Poppins-Medium",
                        "fontSize": 50.5,
                        "fontWeight": "600",
                        "lineHeight": 76,
                      },
                      undefined,
                      Object {
                        "color": "#ffffff",
                      },
                    ],
                  ]
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "rgba(220, 20, 60, 0.09)",
              "borderColor": "#DC143C",
              "borderRadius": 6.5,
              "borderWidth": 0.5,
              "marginTop": 55.5,
              "opacity": 1,
              "paddingHorizontal": 29,
              "paddingVertical": 4.5,
            }
          }
          testID="delete-item"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 26,
                },
                Object {
                  "color": "#DC143C",
                  "fontSize": 32,
                },
              ]
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 55.5,
        }
      }
    />
  </BVLinearGradient>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="delete"
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
